test/src/loaders/workspace_uploader/workspace_uploader_test.py

```diff
@@ -1,5 +1,209 @@
+import os
+import pytest
+import shutil
+import uuid
+from multiprocessing import Queue
+
+from pathlib import Path
+from unittest.mock import Mock
+
+from src.loaders.common import loader_helper
 from src.loaders.workspace_uploader import workspace_uploader
 
 
-def test_noop():
-    assert True
\ No newline at end of file
+@pytest.fixture(scope="function")
+def setup_and_teardown():
+    print('starting workspace uploader tests')
+    tmp_dir = 'result_{}'.format(uuid.uuid4().hex)
+    os.makedirs(tmp_dir)
+
+    # set up sourcedata and collectionssource dirs and soft link them
+    sourcedata_dir = Path(tmp_dir) / "sourcedata" / "NCBI" / "NONE"
+    collection_source_dir = Path(tmp_dir) / "collectionssource"
+    sourcedata_dir.mkdir(parents=True)
+    collection_source_dir.mkdir()
+
+    workspace_id = 12345
+    upload_env_key = "CI"
+    upas = ["12345_58_1", "12345_58_2"]
+    assembly_dir_names = ["GCF_000979855.1", "GCF_000979175.1"]
+    assembly_names = ["GCF_000979855.1_gtlEnvA5udCFS_genomic.fna.gz",
+                      "GCF_000979175.1_gtlEnvA5udCFS_genomic.fna.gz"]
+
+    assembly_dirs, target_files = list(), list()
+    for assembly_dir_name, assembly_name in zip(assembly_dir_names, assembly_names):
+        target_dir_path = sourcedata_dir.joinpath(assembly_dir_name)
+        target_dir_path.mkdir()
+        target_file_path = target_dir_path.joinpath(assembly_name)
+        target_file_path.touch()
+        new_dir_path = collection_source_dir.joinpath(assembly_dir_name)
+        os.symlink(target_dir_path.resolve(), new_dir_path.resolve(), target_is_directory=True)
+        assembly_dirs.append(str(new_dir_path))
+        target_files.append(str(target_file_path))
+
+    yield tmp_dir, assembly_dirs, assembly_names, upas, workspace_id, upload_env_key, target_files
+
+    shutil.rmtree(tmp_dir, ignore_errors=True)
+
+
+def test_get_yaml_file_path(setup_and_teardown):
+    _, assembly_dirs, _, _, _, _, _ = setup_and_teardown
+    yaml_path = workspace_uploader._get_yaml_file_path(assembly_dirs[0])
+
```

Comment:
```suggestion
assert yaml_path.endswith(UPLOADED_YAML)
```

Reply:
I already asserted `os.path.exists(yaml_path)` in line 53. Do I need to add this additional check?

Reply:
👍